packages/labs/observers/src/intersection_controller.ts

```diff
@@ -77,22 +77,22 @@ export class IntersectionController implements ReactiveController {
    * The result of processing the observer's changes via the `callback`
    * function.
    */
-  value?: unknown;
+  value?: T;
   /**
    * Function that returns a value processed from the observer's changes.
    * The result is stored in the `value` property.
    */
-  callback: IntersectionValueCallback = () => true;
+  callback?: IntersectionValueCallback<T>;
```

Comment:
Is removing the default necessary? It's dubious that it's needed, but it is a breaking change and maybe we shouldn't do it unless we have to?

Reply:
It's only a breaking change in the rare situation where someone inspected the code and relied on the default `true | undefined` value (which explicitly typed as `unknown`).

Rationale for removing it was that  it technically defines the default generic type `T` to `T = true` instead of `T = unknown`.
The default callback also seemed like it was for test convenience and I think it might be surprising to find out that the default `value` has  type `true | undefined`, instead of `undefined` .

I may be missing the original reason for the default callback. Was there a benefit to having `value` become `true` after the first observed change?

Reply:
I think the only way to keep the default callback `() => true` is to cast it to `any` thus making type checker happy. Otherwise it's unhappy with the concrete type `T`.

Reply:
Spoke offline - This was not an intended API. Having the callback `undefined` makes more sense as a default and the TS type system has highlighted this.